runtime/libffi/closures.c

```diff
@@ -492,7 +492,8 @@ dlmmap_locked (void *start, size_t length, int prot, int flags, off_t offset)
 	  close (execfd);
 	  goto retry_open;
 	}
-      ftruncate (execfd, offset);
+      if (ftruncate (execfd, offset))
+        return MFAIL;
       return MFAIL;
```

Comment:
Won't make much of a difference, but you could just have an empty-bodied `if` since both paths return the same thing. Ditto for below.

Reply:
Perhaps, but that might also lead the compiler to complain that the return value is ignored: we would have to test your suggestion.

Reply:
I'll test that out now!

Reply:
If this works, I suggest the block should not be hidden at the end of the line, and a comment should make clear that ignoring the return value is intentional:
```c
      if (ftruncate (execfd, offset)) {
        /* ignore any failure */
      }
```

Reply:
There is also `J9_IGNORE_RETURNVAL`: https://github.com/eclipse/omr/blob/87350903300de2d8901b2de5fa564cf9b78d7add/include_core/omrcomp.h#L584

Reply:
That's a nice idea, but I don't think this does (nor should) include that header file, and it would only address the issue with gcc.

Reply:
I can confirm that no warnings appear due to making the `if` bodies empty. I will add those comments indicating the empty bodies are intentional.